packages/wrangler/src/dev.tsx

```diff
@@ -497,93 +498,80 @@ type EsbuildBundle = {
+          // to change the return object's identity
+          setBundle((previousBundle) => {
+            assert(
+              previousBundle,
+              "Rebuild triggered with no previous build available"
+            );
+            return { ...previousBundle, id: previousBundle.id + 1 };
+          });
+        }
+      },
+    };
+
     async function build() {
       if (!destination || !entry) return;
-      const moduleCollector = makeModuleCollector();
-      result = await esbuild.build({
-        entryPoints: [entry],
-        bundle: true,
-        outdir: destination,
-        metafile: true,
-        format: "esm",
-        sourcemap: true,
-        loader: {
-          ".js": "jsx",
-          ".html": "text",
-          ".pem": "text",
-          ".txt": "text",
-        },
-        ...(jsxFactory && { jsxFactory }),
-        ...(jsxFragment && { jsxFragment }),
-        external: ["__STATIC_CONTENT_MANIFEST"],
-        conditions: ["worker", "browser"],
-        plugins: [moduleCollector.plugin],
-        // TODO: import.meta.url
-        watch: {
-          async onRebuild(error) {
-            if (error) console.error("watch build failed:", error);
-            else {
-              // nothing really changes here, so let's increment the id
-              // to change the return object's identity
-              setBundle((previousBundle) => {
-                if (previousBundle === undefined) {
-                  assert.fail(
-                    "Rebuild triggered with no previous build available"
-                  );
-                }
-                return { ...previousBundle, id: previousBundle.id + 1 };
-              });
-            }
-          },
-        },
-      });
-
-      // result.metafile is defined because of the `metafile: true` option above.
-      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-      const metafile = result.metafile!;
-      const outputEntry = Object.entries(metafile.outputs).find(
-        ([_path, { entryPoint }]) => entryPoint === entry
-      ); // assumedly only one entry point
 
-      if (outputEntry === undefined) {
-        throw new Error(
-          `Cannot find entry-point "${entry}" in generated bundle.`
+      const { resolvedEntryPointPath, bundleType, modules, stop } =
+        await bundleWorker(
+          entry,
+          handleAssets,
+          process.cwd(),
+          destination,
+          jsxFactory,
+          jsxFragment,
+          watchMode
         );
-      }
+
+      // Capture the `stop()` method to use as the `useEffect()` destructor.
+      stopWatching = stop;
+
       setBundle({
         id: 0,
         entry,
-        path: outputEntry[0],
-        type: outputEntry[1].exports.length > 0 ? "esm" : "commonjs",
-        exports: outputEntry[1].exports,
-        modules: moduleCollector.modules,
+        path: resolvedEntryPointPath,
+        type: bundleType,
+        modules: modules,
```

Comment:
saw further down, used this style. Change not necessary. 
```suggestion
        modules,
```